### ISSUE TITLE
VB version confusion

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/docs#20147]
The article mentions Visual Basic 15.8, which I can't find in Roslyn source code. They define a LanguageVersion enum

https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/VisualBasic/Portable/LanguageVersion.vb#L12

with the following members:

```vb
    Public Enum LanguageVersion
        [Default] = 0
        VisualBasic9 = 9
        VisualBasic10 = 10
        VisualBasic11 = 11
        VisualBasic12 = 12
        VisualBasic14 = 14
        VisualBasic15 = 15
        VisualBasic15_3 = 1503
        VisualBasic15_5 = 1505
        VisualBasic16 = 1600

        Latest = Integer.MaxValue
    End Enum
```

I *think* that the mentioned improvement for VB 15.8 is actually done for Roslyn that shipped with VS 15.8, and that wasn't under a new language version. This also may mean that a previous PR of mine (https://github.com/dotnet/docs/pull/12884) could be wrong and should be reverted. I must have added that lang version based on what I've seen in this what's new article.

I also think Visual Basic 2017 should be Visual Basic 15.0 (that was the version shipped with Visual Studio 2017)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 272fdffd-2201-990f-387b-d37467e976cf
* Version Independent ID: 06703ef5-e10a-3fdd-da13-0ce99903a5e6
* Content: [What's new for Visual Basic - Visual Basic](https://docs.microsoft.com/en-us/dotnet/visual-basic/getting-started/whats-new)
* Content Source: [docs/visual-basic/getting-started/whats-new.md](https://github.com/dotnet/docs/blob/master/docs/visual-basic/getting-started/whats-new.md)
* Product: **dotnet-visualbasic**
* GitHub Login: @KathleenDollard
* Microsoft Alias: **kdollard**

**Comment by BillWagner** [comment]
ping @KathleenDollard for a recommended resolution

**Comment by Youssef1313** [comment]
@BillWagner, I've updated the issue with another possible issue too.

Could we ask for help here from the compiler team too?

**Comment by adegeo** [comment]
Sent email to @KathleenDollard

**Comment by cston** [comment]
> I think that the mentioned improvement for VB 15.8 is actually done for Roslyn that shipped with VS 15.8, and that wasn't under a new language version.

The [Optimized floating-point to integer conversion](https://docs.microsoft.com/en-us/dotnet/visual-basic/whats-new/#visual-basic-158) change appears to have been made in Visual Studio 15.8 to address https://github.com/dotnet/roslyn/issues/26907. The language version was unchanged at 15.5.
